main.go

```diff
@@ -66,6 +67,8 @@ func main() {
 
 	langserver.GocodeCompletionEnabled = *gocodecompletion
 
+	langserver.FuncSnippetEnabled = *funcSnippetEnabled
```

Comment:
Not directly related to this PR, but I feel like it would help keep the package code more isolated if the above flags were put into `Config` object and passed into `langserver.NewHandler()`.  Is there any interest in this?

Reply:
yes, the organic growth of how this package works has led to this uglyness. Cleaning up sounds great to me.